homeassistant/components/sensor/co2signal.py

```diff
@@ -0,0 +1,133 @@
+"""
+Support for the CO2signal platform.
+For more details about this platform, please refer to the documentation
+"""
+import logging
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import (
+    CONF_TOKEN, CONF_LATITUDE, CONF_LONGITUDE)
+from homeassistant.helpers.config_validation import PLATFORM_SCHEMA
+from homeassistant.helpers.entity import Entity
+
+CONF_REFRESH = "refresh_rate"
+CONF_COUNTRY_CODE = "country_code"
+
+REQUIREMENTS = ['co2signal==0.2']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTRIBUTION = 'Data provided by CO2signal'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_TOKEN, default=None): cv.string,
+    vol.Optional(CONF_COUNTRY_CODE, default=None): cv.string,
+    vol.Optional(CONF_LATITUDE, default=0): cv.latitude,
+    vol.Optional(CONF_LONGITUDE, default=0): cv.longitude,
+    vol.Optional(CONF_REFRESH, default=60): cv.positive_int,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the CO2signal sensor."""
+
+    _LOGGER.debug("Setting up the CO2signal platform")
+
+    token = config.get(CONF_TOKEN)
+    country_code = config.get(CONF_COUNTRY_CODE)
+    refresh_rate = config.get(CONF_REFRESH)
+    lat = None
+    lon = None
+    location_type = None
+
+    # find station ID
+    if country_code is not None:
+        location_type = 'country_code'
+    else:
+        lat = config.get(CONF_LATITUDE)
+        lon = config.get(CONF_LONGITUDE)
+        location_type = 'coordinates'
```

Comment:
Looks like that this could be handled by the validation (`vol.Inclusive` and `vol.Exclusive`).

Reply:
I've updated this part of the code. Is this what you meant?

Reply:
@fabaff I've only used vol.Inclusive as I couldn't parse the default scenario with the HASS coordinates otherwise. Currently it works with all three setups: no coordinates, coordinates and country codes.